lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -654,6 +654,24 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             }
         }
 
+        // handle columns that are removed from changed indexes
+        foreach ($diff->changedIndexes as $chgKey => $chgIndex) {
+            // Dropping primary keys requires to unset autoincrement attribute on the particular column first
+            if ($chgIndex->isPrimary() && $diff->fromTable instanceof Table) {
+
+                foreach ($diff->fromTable->getIndex($chgIndex->getName())->getColumns() as $columnName) {
+                    $column = $diff->fromTable->getColumn($columnName);
+
+                    if ($column->getAutoincrement() === true && in_array($columnName, $chgIndex->getColumns()) === false) {
+                        $column->setAutoincrement(false);
```

Comment:
@andig I know that it is also done for `removedIndexes` and it might be causing the same issues there. So it was done wrong before and should not be done wrong again ;)